src/components/Card.js

```diff
@@ -1,40 +1,74 @@
 export default class Card {
-  constructor({ title, link }, cardSelector, handleImageClick) {
-    this._title = title;
+  constructor(
+    { isLiked, _id, name, link },
+    cardSelector,
+    handleImageClick,
+    handleDeleteClick,
+    handleCardLike
+  ) {
+    this._isLiked = isLiked;
+    this._id = _id;
+    this._name = name;
     this._link = link;
     this._cardSelector = cardSelector;
     this._handleImageClick = handleImageClick;
+    this._handleDeleteClick = handleDeleteClick;
+    this._handleCardLike = handleCardLike;
   }
 
   _setEventListeners() {
     // card like button
     const heart = this._cardElement.querySelector(".heart-button");
     heart.addEventListener("click", () => {
-      this._handleHeartButton();
-      console.log(this._cardElement);
+      this._handleCardLike(
+        this._id,
+        this._heartButtonActivity(),
+        this._handleHeartButton()
```

Comment:
`        this._handleHeartButton()
`

You call it immediately here, but you shouldn't

You need to **toggle likes** only in block `then` after a successful response from the server because if there is an error, the user will not know about that and will see the toggled heart color.

Reply:
This took me a second ! I see now that I was calling it as I passed it in, and therefore it looked like I had working code instead of running after the successful api response.